lib/rules/no-whitespace-before-property.js

```diff
@@ -25,14 +61,20 @@ module.exports = function(context) {
             var prop = node.property;
 
             if (astUtils.isTokenOnSameLine(obj, prop)) {
-                if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
-                    context.report({
-                        node: node,
-                        message: "Unexpected whitespace before property '{{ propName }}'.",
-                        data: {
-                            propName: prop.name
-                        }
-                    });
+                var punc = sourceCode.getTokenBefore(prop);
+
+                if (punc.type !== "Punctuator") {
+                    return;
+                }
+
+                if (punc.value === ".") {
+                    if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
+                        reportError(node, prop);
+                    }
+                } else if (punc.value === "[") {
+                    if (sourceCode.isSpaceBetweenTokens(obj, punc)) {
```

Comment:
They're actually not exactly the same, though it's easy to miss! As per the accepted proposal [here](https://github.com/eslint/eslint/issues/1086), dots are warning if there is space before or after them, while this update aims to only warn for spaces before the enclosing brackets for bracket notation.